### DISCUSSION TITLE
Who is using Mantine DataTable?

### DISCUSSION THREAD

**Discussion by icflorescu** [icflorescu/mantine-datatable#329] [Show and tell]
![Mantine DataTable](https://github.com/icflorescu/mantine-datatable/assets/581999/23ccf41e-732e-4016-82ec-ab76ac17a987)

Hi everyone,

I wanted to thank you all for using Mantine DataTable, for sponsoring me, for being part of the community, for sharing your feedback and contributing to the source code!

I am currently trying to compile a list of startups / projects that are using Mantine DataTable, and I'm planning to place it somewhere at the bottom of the [documentation website](https://icflorescu.github.io/mantine-datatable/) soon.

As you may know, Mantine DataTable is one of my dearest open-source projects and I'm happy to see it nearing 30k downloads per month on `npm`.

I am trying to raise as much awareness as possible, in order to increase the user-base and keep maintaining the project, keep it up-to-date with Mantine core and come up with new features when possible.

I'd also like to add that I'm open for collaborations - if you need help on the front-end and would like to hire my services, feel free to drop me a line! This would greatly contribute to the overall long-term stability of this project.

So, leave your comments below if you're using Mantine DataTable in your project(s), both open-source and commercial, and would like your logo/link on the documentation website.

Happy coding and thank you all for your support!

**Comment by icflorescu** [comment]
List of users already on the docs website:

![Mantine DataTable users](https://github.com/icflorescu/mantine-datatable/assets/581999/f168008f-9ff4-4764-bf0c-40ea61ab6ef9)]

Drop me a line if you think your project/website should be listed here.

**Reply by israelins85** [reply]
Hi Aquarino is starting use, how can I pass the logotype to you?

**Reply by icflorescu** [reply]
Hey @israelins85, thanks for letting me know and apologies for the late reply.
Yes, feel free to send me the logo and a link!

**Comment by mikkurogue** [comment]
I personally use it on some personal projects and at work we also use mantine-datatable. I'll discuss with my colleagues what we can do for you :)

**Reply by icflorescu** [reply]
Hey, @Mikkelzu, thanks for letting me know and for your donation!
Feel free to send me a logo/link if you'd like to appear on the docs website.

**Comment by blentz100** [comment]
@icflorescu Just started using it for a small personal project currently. Thanks for your work - great data table.

**Reply by icflorescu** [reply]
Hey, @blentz100, thank you for letting me know and thanks a lot for your donation!
Feel free to send me a logo/link if you'd like your project to be featured on the docs website at some point.

**Comment by israelins85** [comment]
Hello,

No problem.

I’m finishing the react version off the site, 
in next few days will be released, but the site will be:
http://cloud.aquarino.com.br <http://cloud.aquarino.com.br/>￼

We are a Brazilian Startup, we sell a device to control aquarium.
With the new site we will add new features to be capable to charge per subscription.
When this happens I will discuss with the board the possibility of paying you monthly, 
as we say in Brazil: "Not even a watch works for free."

Thank you for your good work.

Regards,
Israel Lins

**Reply by icflorescu** [reply]
I'd be really grateful each month if you could do that, @israelins85! 🙏😍 
Let me know if you'd like your logo on the new documentation website.

Also, if you're planning to switch to Mantine V7 for your project, here's some good news:

### Mantine DataTable V7.1.0-alpha.5 is here 😅🎉

[![Mantine DataTable V7](https://github.com/icflorescu/mantine-datatable/assets/581999/0be62c66-978e-4ecc-9abe-0dad3254c94d)](https://icflorescu.github.io/mantine-datatable/)

Be advised that it comes with a few [breaking changes](https://github.com/icflorescu/mantine-datatable/blob/main/CHANGELOG.md#710-alpha1-2023-10-30), as expected, since Mantine came up with lots of those.

The amount of effort spent to rewrite the entire documentation website with Next.js 14 was gruesome, but I had to do it in order to make sure the component works in every example.

**Reply by icflorescu** [reply]
Hey, @israelins85, I just remebered what you said about Aquarino.
Can I add it to "companies using Mantine DataTable" at the bottom of the documentation website?

Thanks in advance!

**Comment by israelins85** [comment]
Hi, yes. You can.

—
Regards,
Israel Lins Albuquerque

**Reply by icflorescu** [reply]
Just did, thanks.

**Comment by icflorescu** [comment]
[Mantine DataTable v8.x](https://icflorescu.github.io/mantine-datatable/) released! :partying_face:

**Comment by mikkurogue** [comment]
HI @icflorescu 

we at [bigmile](https://www.bigmile.eu) also use mantine datatable. (i was also a sponsor in the past :D)

just so you know, do what you will with the info

**Reply by icflorescu** [reply]
Would you be ok with me adding your logo and a link on the docs website?

**Reply by mikkurogue** [reply]
For sure!

**Comment by ecatugy** [comment]
@icflorescu  I'm using it, here’s my portal: [labmessage](https://www.labmessage.com.br/). Mantine-datatable is really useful for us.

**Comment by ronnyek** [comment]
I'd really like to be using it. I think we have a bit more in the requirements of being able to handle editing... which could be added without changes to the grid, but I've yet to really grasp how feasible some of what I'd like to do, I can do without a ton of work with this table.

It does 90% of what I want to do out of box and very clean and complete

**Reply by mikkurogue** [reply]
out of curiosity what are you missing in terms of the OOTB experience and what are the main pain points of trying to implement something that may not be an easy setup?

**Reply by ronnyek** [reply]
> out of curiosity what are you missing in terms of the OOTB experience and what are the main pain points of trying to implement something that may not be an easy setup?

well like I said we've got a need to do some inline editing in the grid. I don't think the grid inherently NEEDS to support editing as you can maintain state for edited row/field like I've seen some examples show. I need to also be able to handle keyboard for navigating from this field to that (which understandably probably doesn't exist because editing is not a function of this table).

I'm pretty sure the keyboard navigation could be applied without changes to the grid, but the question is how much work is it to wrap up this component to add that functionality, and still have the end result be compelling?

(none of this is to bash or suggest that native editing at the table level should be added... I'm just saying I think this component is already good, and I'm hoping I can wrap up those remaining gaps (with what we want to do) so that we can use it)

**Reply by mikkurogue** [reply]
Ah yeah, I feel those points. In terms of the inline cell editing I have posted a few examples here and there in some other issues in the project on how I do this but it is far from ideal.

For keyboard navigation, I feel it's probably doable but it very much depends on the state of the table for how you want to navigate. Are you editing, and then do you expect only the row (or column) to be navigateable? If so then usually this can be done if the entire row state is in "edit" mode with the custom solution on top and just by pressing `tab`. Or is navigation only important to traverse cells and then trigger a certain mode? like `hjkl` / `up down left right` to travel and `return` to enter an editing state etc.

I feel that while it's a super cool feature, @icflorescu would probably appreciate then some effort from the community to make this happen as I know he's also swamped with other things. I wouldn't mind collecting more details and helping out with adding this to the table as a native feature, especially because I do believe it will benefit very much from it.

As it stands, everything you want to do is however possible (but not without its pain points and struggles).

**Reply by ronnyek** [reply]
> Ah yeah, I feel those points. In terms of the inline cell editing I have posted a few examples here and there in some other issues in the project on how I do this but it is far from ideal.
> 

I think I am working with your little wrapper, and it does seem to work fairly well. I have issues where I display a date picker and the popup causes the input to blur causing the edit to disappear, and I have an issue where if the field is empty, the div is 0 height, so you cant click it, but I feel like those are minor problems I can figure out.

> For keyboard navigation, I feel it's probably doable but it very much depends on the state of the table for how you want to navigate. Are you editing, and then do you expect only the row (or column) to be navigateable? If so then usually this can be done if the entire row state is in "edit" mode with the custom solution on top and just by pressing `tab`. Or is navigation only important to traverse cells and then trigger a certain mode? like `hjkl` / `up down left right` to travel and `return` to enter an editing state etc.
> 
Ultimately I think I'd use tab to move to next field, next row if you were focused in last column, and maybe hitting enter anywhere would move to next line or something.

Seems like at some point, to display inputs correctly, it may make sense to have something at the cell level to indicate editing, and then the inputs can just get the "unstyled" variant

**Reply by mikkurogue** [reply]
Yeah, so for the date picker issue I think you need a focus trap so both components are focused. I do think that having some intermediate state/tracker is then probably the next logical step. 0 height is I think also a case of just enforcing the default `render()` to always render a div of 100% height (i think). However far from ideal

If i have the time (inbetween a few other projects, work, and of course shitposting on xitter) I could take a look at it all, though i do hope in the meantime that datatable is updated to 8.3.x for the dependencies as that may also give us some more tools from mantine/core
